sdks/java/io/kafka/src/test/java/org/apache/beam/sdk/io/kafka/KafkaIOIT.java

```diff
@@ -79,22 +95,53 @@ public static void setup() throws IOException {
   public void testKafkaIOReadsAndWritesCorrectly() throws IOException {
     writePipeline
         .apply("Generate records", Read.from(new SyntheticBoundedSource(sourceOptions)))
+        .apply("Measure write time", ParDo.of(new TimeMonitor<>(NAMESPACE, "write_time")))
         .apply("Write to Kafka", writeToKafka());
 
-    writePipeline.run().waitUntilFinish();
-
     PCollection<String> hashcode =
         readPipeline
             .apply("Read from Kafka", readFromKafka())
+            .apply("Measure read time", ParDo.of(new TimeMonitor<>(NAMESPACE, "read_time")))
             .apply("Map records to strings", MapElements.via(new MapKafkaRecordsToStrings()))
             .apply("Calculate hashcode", Combine.globally(new HashingFn()).withoutDefaults());
 
     PAssert.thatSingleton(hashcode).isEqualTo(EXPECTED_HASHCODE);
 
+    PipelineResult writeResult = writePipeline.run();
+    writeResult.waitUntilFinish();
+
     PipelineResult readResult = readPipeline.run();
     PipelineResult.State readState =
         readResult.waitUntilFinish(Duration.standardSeconds(options.getReadTimeout()));
     cancelIfNotTerminal(readResult, readState);
+
+    Set<NamedTestResult> metrics = readMetrics(writeResult, readResult);
+    IOITMetrics.publish(
+        testId, timestamp, options.getBigQueryDataset(), options.getBigQueryTable(), metrics);
+  }
+
+  private Set<NamedTestResult> readMetrics(PipelineResult writeResult, PipelineResult readResult) {
+    Function<MetricsReader, NamedTestResult> writeTimeSupplier =
+        reader -> {
+          long start = reader.getStartTimeMetric("write_time");
```

Comment:
Please, extract "write_time" as a constant value and use this value over the test instead.